game/src/contrib/components/HealthComponent.java

```diff
@@ -76,12 +78,14 @@ public HealthComponent(@DSLContextMember(name = "entity") Entity entity) {
     }
 
     /**
-     * Adds damage, which is accounted for by the system
+     * Adds damage, which is accounted for by the system and creates a damage popup for the hero.
      *
      * @param damage Damage that should be inflicted
      */
     public void receiveHit(Damage damage) {
         damageToGet.add(damage);
+        if (this.getEntity().equals(Game.getHero().orElseThrow()))
```

Comment:
Mir missfällt, dass hier im Component eine Unterscheidung gemacht wird, ob es zum Hero gehört oder nicht.

Kann man das Konstrukt etwas umdrehen? Anstatt dass sich die Komponente meldet, sollte das UI die Komponente überwachen, also in jedem Frame prüfen, ob sich etwas an der Komponente geändert hat.